src/devtools/client/debugger/src/actions/sources/select.ts

```diff
@@ -132,9 +132,10 @@ export function selectLocation(
     if (location.sourceUrl && location.sourceUrl !== source?.url) {
       await ThreadFront.ensureAllSources();
       const state = getState();
-      const sourceId = getSourceToDisplayForUrl(state, location.sourceUrl)!.id;
-      source = getSourceDetails(state, sourceId);
-      location = { ...location, sourceId };
+      source = getSourceToDisplayForUrl(state, location.sourceUrl);
+      if (source) {
+        location = { ...location, sourceId: source.id };
```

Comment:
Is the `!source` case correct? We'll end up calling...
```js
    dispatch(locationSelected({ location, source }));
```
...with a `PartialLocation` rather than a `Location`. That seems like it might cause trouble in other places?

Reply:
No, we'll end up calling `return dispatch(clearSelectedLocation());`